table/sst_file_writer.cc

```diff
@@ -462,6 +494,36 @@ Status SstFileWriter::Finish(ExternalSstFileInfo* file_info) {
 
   if (file_info != nullptr) {
     *file_info = r->file_info;
+    // Remove user-defined timestamps from external file metadata too when they
+    // should not be persisted.
+    assert(r->file_info.smallest_key.empty() ==
+           r->file_info.largest_key.empty());
+    assert(r->file_info.smallest_range_del_key.empty() ==
+           r->file_info.largest_range_del_key.empty());
+    if (r->strip_timestamp) {
+      if (!r->file_info.smallest_key.empty()) {
+        Slice smallest_key = r->file_info.smallest_key;
+        Slice largest_key = r->file_info.largest_key;
+        assert(smallest_key.size() >= r->ts_sz);
+        assert(largest_key.size() >= r->ts_sz);
+        file_info->smallest_key.assign(smallest_key.data(),
+                                       smallest_key.size() - r->ts_sz);
+        file_info->largest_key.assign(largest_key.data(),
+                                      largest_key.size() - r->ts_sz);
```

Comment:
`*file_info` at this point contains a copy of `r->file_info`, right? If so, we could use `std::string::resize()` to chop off the timestamps in place.

Reply:
That's a good idea!